### DISCUSSION TITLE
Separate endpoint location for OpenAPI

### DISCUSSION THREAD

**Discussion by azzamsa** [poem-web/poem#223] [Q&A] [answered]
Hi.

I want to add OpenApi feature for my current Poem app. Each endpoint is located in its own module (/api/queries/endpoint_name.rs).
Such as:

``` shell
❯ exa --git-ignore -T meta
meta
├── api
│  └── queries
│     ├── find_meta.rs


❯ exa --git-ignore -T health/
health
├── api
│  └── queries
│     ├── find_health.rs
```

I did the same with async-graphql and I can use [MergedObject](https://docs.rs/async-graphql/latest/async_graphql/derive.MergedObject.html) to merge my Quries and Mutations that scattered in each own module.

```rust
// src/health/resolver.rs
#[Object]
impl HealthQuery {
    pub async fn health(&self, ctx: &Context<'_>) -> FieldResult<Health> {
    ...
    }
}

// src/meta/resolver.rs
#[Object]
impl MetaQuery {
    pub async fn meta(&self, ctx: &Context<'_>) -> FieldResult<Meta> {
    ...
    }
}

// main.rs
#[derive(MergedObject, Default)]
pub struct Query(
    MetaQuery,
    HealthQuery,
);
```

Could I do something simmiar?
I don't want all my endpoint located in one place like this https://github.com/poem-web/poem/blob/9326e972add8af63a9456eeb9c6d4aa17b543307/examples/openapi/users-crud/src/main.rs#L84.

Thanks in advance

**Comment by sunli829** [comment, marked as answer]
You can combine multiple APIs.

https://github.com/poem-web/poem/blob/9326e972add8af63a9456eeb9c6d4aa17b543307/examples/openapi/combined-apis/src/main.rs#L41

**Reply by azzamsa** [reply]
Worked like a charm.
